examples/thermally-coupled-mpi.py

```diff
@@ -85,6 +79,82 @@ class MyRuntimeError(RuntimeError):
+
+    # Insert the interface boundaries for computing the gradient
+    from mirgecom.multiphysics.thermally_coupled_fluid_wall import \
+        add_interface_boundaries_no_grad
+    fluid_all_boundaries_no_grad, wall_all_boundaries_no_grad = \
+        add_interface_boundaries_no_grad(
+            dcoll,
+            gas_model,
+            fluid_dd, wall_dd,
+            fluid_state, wall_kappa, wall_temperature,
+            fluid_boundaries, wall_boundaries)
+
+    # Get the operator fluid states
+    from mirgecom.gas_model import make_operator_fluid_states
+    fluid_operator_states_quad = make_operator_fluid_states(
+        dcoll, fluid_state, gas_model, fluid_all_boundaries_no_grad,
+        quadrature_tag, dd=fluid_dd)
+
+    # Compute the temperature gradient for both subdomains
+    from mirgecom.navierstokes import grad_t_operator as fluid_grad_t_operator
+    from mirgecom.diffusion import grad_operator as wall_grad_t_operator
+    fluid_grad_temperature = fluid_grad_t_operator(
+        dcoll, gas_model, fluid_all_boundaries_no_grad, fluid_state,
+        time=time, quadrature_tag=quadrature_tag,
+        dd=fluid_dd, operator_states_quad=fluid_operator_states_quad)
+    wall_grad_temperature = wall_grad_t_operator(
+        dcoll, wall_kappa, wall_all_boundaries_no_grad, wall_temperature,
+        quadrature_tag=quadrature_tag, dd=wall_dd)
+
+    # Insert boundaries for the fluid-wall interface, now with the temperature
+    # gradient
+    from mirgecom.multiphysics.thermally_coupled_fluid_wall import \
+        add_interface_boundaries
+    fluid_all_boundaries, wall_all_boundaries = \
+        add_interface_boundaries(
+            dcoll,
+            gas_model,
+            fluid_dd, wall_dd,
+            fluid_state, wall_kappa, wall_temperature,
+            fluid_grad_temperature, wall_grad_temperature,
+            fluid_boundaries, wall_boundaries)
+
+    # Compute the subdomain NS/diffusion operators using the augmented boundaries
+    from mirgecom.navierstokes import ns_operator
+    from mirgecom.diffusion import diffusion_operator
+    ns_result = ns_operator(
+        dcoll, gas_model, fluid_state, fluid_all_boundaries,
+        time=time, quadrature_tag=quadrature_tag, dd=fluid_dd,
+        return_gradients=return_gradients,
+        operator_states_quad=fluid_operator_states_quad,
+        grad_t=fluid_grad_temperature)
+    diffusion_result = diffusion_operator(
+        dcoll, wall_kappa, wall_all_boundaries, wall_temperature,
+        quadrature_tag=quadrature_tag, return_grad_u=return_gradients, dd=wall_dd,
+        grad_u=wall_grad_temperature)
+
+    if return_gradients:
+        fluid_rhs, fluid_grad_cv, fluid_grad_temperature = ns_result
+        wall_rhs, wall_grad_temperature = diffusion_result
+        return (
+            fluid_rhs, wall_rhs, fluid_grad_cv, fluid_grad_temperature,
+            wall_grad_temperature)
+    else:
+        return ns_result, diffusion_result
+
+
```

Comment:
I like this spirit of this change set, but I'm slightly concerned that this causes most of what is highlighted here to be repeated in every coupled driver, adding to an already onerous driver maintenance situation.

Can any of this be offered as a canned interface?

Reply:
I was thinking the exact same thing. I would keep the operator instead of deprecating it.

Reply:
Hmm. Well, the primary motivation for removing it was that it was proving difficult to come up with a one-size-fits-all operator definition (because simulations may have different numbers of volumes, or they may want to precompute things like the operator states and/or gradients). Seemed the most reasonable to just have the driver writer decide how they want to call the operators. I could leave this in as a default/example, but I think I would want to rename it to emphasize that (`basic_ns_heat_operator` or something?). Thoughts?

Reply:
I suspected this would be the case.   The `basic` operator idea is fine for me; on the other hand the example also helps with this.  Aren't the stand-burner case and the prediction case identical in this respect?   (er couldn't they be?).  If so, I would actually prefer for this function to be centralized for both those drivers, and only re-implemented in each driver that needs to re-implement them.

Also, these are just suggestions - nothing show-stopping in my opinion.

Reply:
For the prediction/burner we also need the NS coupling, not only the heat equation coupling. Having the triple coupling is what motivated this PR. So I wouldn't worry in making this particular coupling too focused on the more complex drivers we have.

Reply:
It seems any way we slice it, we're going to have many versions of this coupled operator.  The more i read it, the shorter it gets :)